React/Modules/RCTUIManager.m

```diff
@@ -1183,6 +1201,16 @@ - (void)flushUIBlocks
     RCTProfileBeginFlowEvent();
     dispatch_async(dispatch_get_main_queue(), ^{
       RCTProfileEndFlowEvent();
+
+      // Copy the observers set to avoid having to lock for too long while running ui blocks.
```

Comment:
More importantly this avoids needing to think about concurrent modification of the observers set while the observers run (i.e. when an observer adds or removes other observers)